### PR TITLE
drivers: sensor: bmm150: Add trigger support for bmm150 magnetometer …

### PR DESCRIPTION
**Pull request by Cherish-Gww**
…sensor

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensor/sensors_trigger_*.conf`

File: dts/bindings/sensor/bosch,bmm150.yaml [hunk 1]
```diff
@@ -4,3 +4,7 @@
 # Common fields for BMM150
 
 include: sensor-device.yaml
+
+properties:
+  drdy-gpios:
+    type: phandle-array
```
Comment (MaureenHelm) [inline dts/bindings/sensor/bosch,bmm150.yaml:10]: Please add a description

File: tests/drivers/build_all/sensor/sensors_trigger_global.conf [hunk 2]
```diff
@@ -49,3 +49,4 @@ CONFIG_TMP007_TRIGGER_GLOBAL_THREAD=y
 CONFIG_VCNL4040_TRIGGER_GLOBAL_THREAD=y
 CONFIG_WSEN_HIDS_TRIGGER_GLOBAL_THREAD=y
 CONFIG_WSEN_TIDS_TRIGGER_GLOBAL_THREAD=y
+CONFIG_BMM150_TRIGGER_GLOBAL_THREAD=y
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/sensors_trigger_global.conf:52]: Move up to preserve alpha order

File: tests/drivers/build_all/sensor/sensors_trigger_none.conf [hunk 3]
```diff
@@ -49,3 +49,4 @@ CONFIG_TMP007_TRIGGER_NONE=y
 CONFIG_VCNL4040_TRIGGER_NONE=y
 CONFIG_WSEN_HIDS_TRIGGER_NONE=y
 CONFIG_WSEN_TIDS_TRIGGER_NONE=y
+CONFIG_BMM150_TRIGGER_NONE=y
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/sensors_trigger_none.conf:52]: Move up to preserve alpha order

File: tests/drivers/build_all/sensor/sensors_trigger_own.conf [hunk 4]
```diff
@@ -46,3 +46,4 @@ CONFIG_TMP007_TRIGGER_OWN_THREAD=y
 CONFIG_VCNL4040_TRIGGER_OWN_THREAD=y
 CONFIG_WSEN_HIDS_TRIGGER_OWN_THREAD=y
 CONFIG_WSEN_TIDS_TRIGGER_OWN_THREAD=y
+CONFIG_BMM150_TRIGGER_OWN_THREAD=y
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/sensors_trigger_own.conf:49]: Move up to preserve alpha order

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please address the failed twister check:
```
In file included from /__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c:15:
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150.h:175:22: error: duplicate member ‘sem’
  175 |         struct k_sem sem;
      |                      ^~~
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c: In function ‘bmm150_handle_interrupts’:
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c:24:19: error: ‘struct bmm150_data’ has no member named ‘drdy_handler’; did you mean ‘drdy_handle’?
   24 |         if (data->drdy_handler) {
      |                   ^~~~~~~~~~~~
      |                   drdy_handle
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c:25:23: error: ‘struct bmm150_data’ has no member named ‘drdy_handler’; did you mean ‘drdy_handle’?
   25 |                 data->drdy_handler(dev, data->drdy_trigger);
      |                       ^~~~~~~~~~~~
      |                       drdy_handle
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c: In function ‘bmm150_trigger_set’:
/__w/zephyr/zephyr/drivers/sensor/bmm150/bmm150_trigger.c:102:15: error: ‘struct bmm150_data’ has no member named ‘drdy_handler’; did you mean ‘drdy_handle’?
  102 |         data->drdy_handler = handler;
      |               ^~~~~~~~~~~~
      |               drdy_handle
```

**Review by MaureenHelm — DISMISSED** [review]

**Comment by Cherish-Gww** [comment]
@MaureenHelm Do you still have any issues with this patch ?

**Review by MaureenHelm — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
@Cherish-Gww please rebase

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**